core/tests/protractor_utils/ContributorDashboardPage.js

```diff
@@ -17,22 +17,221 @@
+    await waitFor.visibilityOf(
+      usernameContainer, 'Username takes too much time to load');
+  };
 
   this.getTranslateTextTab = function() {
     return new ContributorDashboardTranslateTextTab
       .ContributorDashboardTranslateTextTab();
   };
+
+  this.waitForOpportunitiesToLoad = async function() {
+    await waitFor.invisibilityOf(
+      opportunityLoadingPlaceholder,
+      'Opportunity placeholders take too long to become invisible.');
+  };
+
+  this.waitForQuestionSuggestionReviewModalToAppear = async function() {
+    await waitFor.visibilityOf(
+      questionReviewModalHeader,
+      'Question suggestion review modal takes too long to appear');
+  };
+
+  this.expectUserToBeTranslationReviewer = async function(language) {
+    await waitFor.visibilityOf(
+      reviewRightsDiv, 'User does not have rights to review translation');
+
+    var translationReviewRightsElement = element(by.css(
+      '.protractor-test-translation-' + language + '-reviewer'));
+    await waitFor.visibilityOf(
+      translationReviewRightsElement,
+      'User does not have rights to review translation in language: ' + language
+    );
+  };
+
+  var _expectUserToBeReviewer = async function(
+      reviewCategory, langaugeDescription = null) {
+    await waitFor.visibilityOf(
+      reviewRightsDiv,
+      'Review rights div is not visible, so user does not have rights to ' +
+      'review ' + reviewCategory);
+
+    var reviewRightsElementClassName = ('.protractor-test-' + reviewCategory);
+    if (langaugeDescription !== null) {
+      reviewRightsElementClassName += '-' + langaugeDescription;
+    }
+    reviewRightsElementClassName += '-reviewer';
+
+    var reviewRightsElement = element(by.css(reviewRightsElementClassName));
+    await waitFor.visibilityOf(
+      reviewRightsElement,
+      'Review right element ' + reviewCategory + ' is not visible');
+  };
+
+  this.expectUserToBeTranslationReviewer = async function(langaugeDescription) {
+    await _expectUserToBeReviewer('translation', langaugeDescription);
+  };
+
+  this.expectUserToBeVoiceoverReviewer = async function(langaugeDescription) {
+    await _expectUserToBeReviewer('voiceover', langaugeDescription);
+  };
+
+  this.expectUserToBeQuestionReviewer = async function() {
+    await _expectUserToBeReviewer('question');
+  };
+
+  this.expectNumberOfOpportunitiesToBe = async function(number) {
+    var opportunityCount = (await _getOpportunityElements()).length;
+    expect(opportunityCount).toBe(number);
+  };
+
+  this.expectEmptyOpportunityAvailabilityMessage = async function() {
+    await waitFor.visibilityOf(
+      opportunityListEmptyAvailabilityMessage,
+      'Opportunity list is not empty');
+  };
+
+  var _getOpportunityElements = async function() {
+    var opportunityCount = await opportunityListItems.count();
+
+    var opportunityElements = [];
+    for (var i = 0; i < opportunityCount; i++) {
+      var opportunityElement = await opportunityListItems.get(i);
```

Comment:
Can we check that these opportunity elements have loaded first? It looks like there might not be any, so maybe we could check that some other element that loads at around the same time has loaded?

Reply:
On line 135 I've counted the number of opportunityListItems and then I'm lopping through the counts. Also, how should I validate multiple elements visibility here? (I'm not sure!)

Reply:
Since there might not be any elements, I don't think you can wait for opportunityListItems directly. Instead, you could find another element on the page that loads around the same time? Alternatively, you could waitFor oportunityListItems.first() and catch any exceptions. If there is an exception thrown, skip the for loop since no elements exist. Would that work?

Reply:
Done! [Added assertion for an alternative element which loads inside the one element of the list.)